cluster_test.go

```diff
@@ -120,6 +127,127 @@ func TestListParams(t *testing.T) {
+	server := &http.Server{}
+	server.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		url := r.Header["X-Forwarded-Proto"][0] + "://" + r.Header["X-Forwarded-Host"][0] + r.URL.RequestURI()
+		signatureHeader := r.Header["X-Twilio-Signature"]
+		r.ParseForm()
+		params := make(map[string]string)
+		for k, v := range r.PostForm {
+			params[k] = v[0]
+		}
+		requestValidator := twilio.NewRequestValidator(os.Getenv("TWILIO_AUTH_TOKEN"))
+		if len(signatureHeader) != 0 {
+			channel <- requestValidator.Validate(url, params, r.Header["X-Twilio-Signature"][0])
+		} else {
+			channel <- false
+		}
+	})
+	return server
+}
+
+func createStudioFlowParams(url string, method string) *StudioV2.CreateFlowParams {
+	jsonStr := fmt.Sprintf(`{
+		"description": "Studio Flow",
+		"states": [
+		  {
+			"name": "Trigger",
+			"type": "trigger",
+			"transitions": [
+			  {
+				"next": "httpRequest",
+				"event": "incomingRequest"
+			  }
+			],
+			"properties": {
+			}
+		  },
+		  {
+			"name": "httpRequest",
+			"type": "make-http-request",
+			"transitions": [],
+			"properties": {
+			  "method": "%s",
+			  "content_type": "application/x-www-form-urlencoded;charset=utf-8",
+			  "url": "%s"
+			}
+		  }
+		],
+		"initial_state": "Trigger",
+		"flags": {
+		  "allow_concurrent_calls": true
+		}
+	  }`, method, url)
+
+	var definition interface{}
+	_ = json.Unmarshal([]byte(jsonStr), &definition)
+
+	params := &StudioV2.CreateFlowParams{
+		Definition: &definition,
+	}
+	params.SetFriendlyName("Go Cluster Test Flow")
+	params.SetStatus("published")
+	return params
+}
+
+func createStudioExecutionParams() *StudioV2.CreateExecutionParams {
+	executionParams := &StudioV2.CreateExecutionParams{}
+	executionParams.SetTo("To")
+	executionParams.SetFrom("From")
+	return executionParams
+}
+
+func executeFlow(t *testing.T, flowSid string) {
+	_, exeErr := testClient.StudioV2.CreateExecution(flowSid, createStudioExecutionParams())
+	if exeErr != nil {
+		t.Fatal("Error with Studio Execution Creation: ", exeErr)
+	}
+}
+
+func requestValidation(t *testing.T, method string) {
+	//Invoke Localtunnel
+	listener, ltErr := localtunnel.Listen(localtunnel.Options{})
+	if ltErr != nil {
+		t.Fatal("Error with Localtunnel: ", ltErr)
+	}
+	//Create Validation Server & Listen
+	channel := make(chan bool)
+	server := createValidationServer(channel)
+	go server.Serve(listener)
+
+	//Extra time for server to set up
+	time.Sleep(1 * time.Second)
+
+	//Create Studio Flow
+	params := createStudioFlowParams(listener.URL(), method)
+	resp, flowErr := testClient.StudioV2.CreateFlow(params)
+	if flowErr != nil {
+		t.Fatal("Error with Studio Flow Creation: ", flowErr)
+	}
+	flowSid := *resp.Sid
+	executeFlow(t, flowSid)
+
+	//Await for Request Validation
+	afterCh := time.After(5 * time.Second)
+	select {
+	case validate := <-channel:
+		assert.True(t, validate)
+	case <-afterCh:
+		t.Fatal("No request was sent to validation server")
+	}
```

Comment:
Relying on a timeout of 5 seconds if no request hit the validation server. This is a workaround as StudioV2.FetchExecutionContext was not working as expected.